RootTests: libCPyCppyy.so: undefined symbol: Py_HashPointer

### Check duplicate issues.

- [x] Checked for duplicates

### Description

There are multiple tests failing with this error message in the ROOT master branch

```
-- TEST COMMAND -- 
cd /build/jenkins/workspace/lcg_nightly_pipeline/build/projects/ROOT-HEAD/src/ROOT-HEAD-build/roottest/main
/usr/bin/timeout -s USR2 270s /build/jenkins/workspace/lcg_nightly_pipeline/build/projects/ROOT-HEAD/src/ROOT-HEAD-build/bin/rootrm -r copy_replace_recursive.root
-- BEGIN TEST OUTPUT --

-- END TEST OUTPUT --
-- BEGIN TEST ERROR --
Traceback (most recent call last):
  File "/build/jenkins/workspace/lcg_nightly_pipeline/build/projects/ROOT-HEAD/src/ROOT-HEAD-build/bin/rootrm", line 10, in <module>
    import cmdLineUtils
  File "/build/jenkins/workspace/lcg_nightly_pipeline/build/projects/ROOT-HEAD/src/ROOT-HEAD-build/lib/cmdLineUtils.py", line 79, in <module>
    import ROOT
  File "/build/jenkins/workspace/lcg_nightly_pipeline/build/projects/ROOT-HEAD/src/ROOT-HEAD-build/lib/ROOT/__init__.py", line 20, in <module>
    import cppyy
  File "/build/jenkins/workspace/lcg_nightly_pipeline/build/projects/ROOT-HEAD/src/ROOT-HEAD-build/lib/cppyy/__init__.py", line 74, in <module>
    from ._cpython_cppyy import *
  File "/build/jenkins/workspace/lcg_nightly_pipeline/build/projects/ROOT-HEAD/src/ROOT-HEAD-build/lib/cppyy/_cpython_cppyy.py", line 26, in <module>
    import cppyy.libcppyy as _backend
ImportError: /build/jenkins/workspace/lcg_nightly_pipeline/build/projects/ROOT-HEAD/src/ROOT-HEAD-build/lib/libCPyCppyy.so: undefined symbol: Py_HashPointer

-- END TEST ERROR --
CMake Error at /build/jenkins/workspace/lcg_nightly_pipeline/build/projects/ROOT-HEAD/src/ROOT/HEAD/cmake/modules/RootTestDriver.cmake:186 (message):
  got exit code 1 but expected 0
```

roottest-main-NameCyclesRootmv 
roottest-main-RootcpReadRemoteClean 
roottest-main-RootcpReplaceRecursiveClean 
roottest-main-SimpleRootcp1Clean 
roottest-main-SimpleRootcp2Clean 
roottest-main-SimpleRootcp3Clean 
roottest-main-SimpleRootcp4Clean 
roottest-main-SimpleRootcp5Clean 
roottest-main-SimpleRootmkdir1 
roottest-main-SimpleRootmkdir2 
roottest-main-SimpleRootmkdir3 
roottest-main-SimpleRootmv1 
roottest-main-SimpleRootmv2 
roottest-main-SimpleRootmv3 
roottest-main-SimpleRootmv4 
roottest-main-SimpleRootmv5 
roottest-main-SimpleRootrm1 
roottest-main-SimpleRootrm2 
roottest-main-SimpleRootrm3 
roottest-root-multicore-ttree_read_imt

See: https://lcgapp-services.cern.ch/cdash/viewTest.php?onlyfailed&buildid=142202

On January 24 (midnight) these were not yet failing.
We are building against python 3.13.11

### Reproducer

build from source against, run ctest

### ROOT version

ROOT master

### Installation method

from source

### Operating system

Alma Linux 9

### Additional context

Python 3.13.11

--- a/roottest/main/CMakeLists.txt
+++ b/roottest/main/CMakeLists.txt
@@ -6,16 +6,15 @@ configure_file(subdirs.root . COPYONLY)
 configure_file(nested.root . COPYONLY)
 configure_file(RNTuple.root . COPYONLY)
 
+# We should explicitly use the Python executable from the Python version that
+# was used to build ROOT. Otherwise, we risk picking up a different Python
+# version at test time.
+set(PY_TOOLS_PREFIX ${Python3_EXECUTABLE} ${ROOTSYS}/bin)
+set(TOOLS_PREFIX ${ROOTSYS}/bin)
+
 if(MSVC)
-    # the command line tools works fine in the Windows command prompt but
-    # not from CTest, so let's add Python.exe and the .py file extension
-    set(PY_TOOLS_PREFIX ${Python3_EXECUTABLE} ${ROOTSYS}/bin)
-    set(TOOLS_PREFIX ${ROOTSYS}/bin)
     set(pyext .py)
     set(exeext .exe)
-else()
-    set(PY_TOOLS_PREFIX ${ROOTSYS}/bin)
-    set(TOOLS_PREFIX ${ROOTSYS}/bin)
 endif()
 
 ############################## ROOLS TESTS ##############################

